orchagent/response_publisher.cpp

```diff
@@ -0,0 +1,170 @@
+#include "response_publisher.h"
+
+#include <fstream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "timestamp.h"
+
+extern bool gResponsePublisherRecord;
+extern bool gResponsePublisherLogRotate;
+extern std::ofstream gResponsePublisherRecordOfs;
+extern std::string gResponsePublisherRecordFile;
+
+namespace {
+
+// Returns the component string that we need to prepend for sending the error
+// message.
+// Returns an empty string if the status is OK.
+// Returns "[SAI] " if the ReturnCode is generated from a SAI status code.
+// Else, returns "[OrchAgent] ".
+std::string PrependedComponent(const ReturnCode& status) {
+  constexpr char* kOrchagentComponent = "[OrchAgent] ";
+  constexpr char* kSaiComponent = "[SAI] ";
+  if (status.ok()) {
+    return "";
+  }
+  if (status.isSai()) {
+    return kSaiComponent;
+  }
+  return kOrchagentComponent;
+}
+
+void PerformLogRotate() {
+  if (!gResponsePublisherLogRotate) {
```

Comment:
Suggest to follow the existing swss coding style - 4 space indent and { -> new line